src/layout/MasterLayout.cpp

```diff
@@ -30,6 +30,28 @@ int CHyprMasterLayout::getMastersOnWorkspace(const int& ws) {
     return no;
 }
 
+SMasterWorkspaceData* CHyprMasterLayout::getMasterWorkspaceData(const int& ws) {
+    for (auto& n : m_lMasterWorkspacesData) {
+        if (n.workspaceID == ws)
+            return &n;
+    }
+
+    //create on the fly if it doesn't exist yet
+    const auto PWORKSPACEDATA = &m_lMasterWorkspacesData.emplace_back();
+    PWORKSPACEDATA->workspaceID = ws;
+    const auto orientation = g_pConfigManager->getString("master:orientation");
```

Comment:
see https://wiki.hyprland.org/Contributing-and-Debugging/PR-Guidelines/#some-code-faq

Reply:
Addressed now in 3d3431223181afc716ae46d7059dfc2ccadb5b5c , does this look better?

Reply:
yes